plugins/groups_manager.py

```diff
@@ -0,0 +1,307 @@
+            demands.sack_activation = True
+            demands.available_repos = True
+            demands.load_system_repo = False
+
+        # handle --merge option (shortcut to --load and --save the same file)
+        if self.opts.merge:
+            self.opts.load.insert(0, self.opts.merge)
+            self.opts.save.append(self.opts.merge)
+
+        # check that group is specified when editing is attempted
+        if (self.opts.description
+                or self.opts.display_order
+                or self.opts.translated_name
+                or self.opts.translated_description
+                or self.opts.user_visible is not None
+                or self.opts.packages):
+            if not self.opts.id and not self.opts.name:
+                raise dnf.cli.CliError(
+                    _("Can't edit group without specifying it (use --id or --name)"))
+
+    def load_input_files(self):
+        """
+        Loads all input xml files.
+        Returns True if at least one file was successfuly loaded
+        """
+        for file_name in self.opts.load:
+            file_comps = libcomps.Comps()
+            try:
+                if file_name.endswith('.gz'):
+                    # libcomps does not support gzipped files - decompress to temporary
+                    # location
+                    with gzip.open(file_name) as gz_file:
+                        with tempfile.NamedTemporaryFile() as temp_file:
+                            shutil.copyfileobj(gz_file, temp_file)
+                            # The xml written to temp_file  might remain in buffers.
+                            # Force writing it to disk.
+                            temp_file.file.flush()
+                            os.fsync(temp_file.file.fileno())
+                            file_comps.fromxml_f(temp_file.name)
+                else:
+                    file_comps.fromxml_f(file_name)
+            except (IOError, OSError, libcomps.ParserError) as err:
+                # gzip module raises OSError on reading from malformed gz file
+                logger.error(_("Can't load file \"{}\": {}").format(file_name, err))
+                # get_last_errors() output often contains duplicit lines, remove them
+                seen = set()
+                for error in file_comps.get_last_errors():
+                    if error in seen:
+                        continue
+                    logger.error(error.strip())
+                    seen.add(error)
+                raise dnf.exceptions.Error()
+            else:
+                self.comps += file_comps
+
+    def save_output_files(self):
+        for file_name in self.opts.save:
+            try:
+                # xml_f returns a list of errors / log entries
+                errors = self.comps.xml_f(file_name, xml_options=COMPS_XML_OPTIONS)
+            except libcomps.XMLGenError as err:
+                errors = [err]
+            if errors:
+                logger.error(''.join(errors).strip())
```

Comment:
Does this not result in an error (along with a non-zero exit code)? Seems like there should be an exception raised. Similar to the load method, I would put the error description into the exception if there is just one. If there are multiple I would log them and then raise an exception like "Error saving file FILENAME".

Reply:
I've changed it so that the exit code 1 is returned (along with details of the error logged) when saving the output file fails.